src/components/sidebar/CollapseComponet.js

```diff
@@ -0,0 +1,48 @@
+import {
+  Collapse,
+  Checkbox,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+} from '@mui/material';
+import { default as Popover } from './PopoeverComponent';
+import ExpandLess from '@mui/icons-material/ExpandLess';
+import ExpandMore from '@mui/icons-material/ExpandMore';
+import { useState } from 'react';
+
+export default function CollapseComponet({
+  collapseHeader,
+  collapseItems,
+  services,
+}) {
+  const [collapseOpen, setCollapseOpen] = useState(false);
+
+  const handleCollapse = (e) => {
+    setCollapseOpen((previousCollapseOpen) => !previousCollapseOpen);
+  };
+
+  return (
+    <>
+      <ListItemButton onClick={handleCollapse}>
+        <ListItemText primary={collapseHeader} />
+        {Object.keys(services).length === 0 ? null : (
```

Comment:
I think`{Object.keys(services).length > 0 &&  <Popover services={services} />}` might be a better use case here.